bindings/python/pymongoarrow/api.py

```diff
@@ -336,6 +337,8 @@ def write(collection, tabular):
             collection.insert_many(cur_batch)
         except BulkWriteError as bwe:
             raise ArrowWriteError(_transform_bwe(dict(bwe.details), cur_offset)) from bwe
+        except pymongo.errors.PyMongoError as pme:
+            raise ArrowWriteError({"cause": pme, "index": cur_offset}) from pme
```

Comment:
This makes for a somewhat unwieldy api since we raise ArrowWriteError in two different forms. Is there any way to unify the two apis?

Also I think we can get rid of "cause" since users can already access `exc. __cause__ ` set by the `raise ArrowWriteError(...) from pme`. See:
https://docs.python.org/3/tutorial/errors.html#exception-chaining  and https://docs.python.org/3/library/exceptions.html#exception-context for more info about chaining.

Reply:
If we can get rid of cause, then we can raise an ArrowWriteError of the same form as the bulk write error (ie. with ``"writeErrors": {"errmsg": exc._message, "index":cur_offset}`` this form)

Reply:
Great!